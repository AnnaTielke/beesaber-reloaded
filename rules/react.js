module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
