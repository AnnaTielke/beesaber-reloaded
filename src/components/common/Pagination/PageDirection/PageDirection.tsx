import React from 'react';

type Props = {
  direction: 'Next' | 'Previous';
  onClick: (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
};

const PageDirection = ({ onClick, direction }: Props): JSX.Element => (
  <li className="page-item">
    <a className="page-link" href="/#" aria-label={direction} onClick={onClick}>
      <span aria-hidden="true">&laquo;</span>
      <span className="sr-only">{direction}</span>
    </a>
  </li>
);

export default PageDirection;
