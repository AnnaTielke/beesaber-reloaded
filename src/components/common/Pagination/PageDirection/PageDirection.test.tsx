import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import PageDirection from './PageDirection';

describe('PageDirection', () => {
  const mockOnClick = jest.fn();

  afterEach(() => jest.resetAllMocks());

  it.each(['Next', 'Previous'] as const)(
    'should match the snapshot when direction is = %s',
    (direction) => {
      const { container } = render(
        <PageDirection direction={direction} onClick={mockOnClick} />
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it('should call onClick when clicking the direction', () => {
    render(<PageDirection direction={'Next'} onClick={mockOnClick} />);

    const pageLink = screen.getByRole('link', { name: 'Next' });

    userEvent.click(pageLink);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
