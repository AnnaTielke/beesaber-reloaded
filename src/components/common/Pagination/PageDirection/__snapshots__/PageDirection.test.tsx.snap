// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDirection should match the snapshot when direction is = Next 1`] = `
.c0 {
  background-color: rgba(255,255,255,1)!important;
  font-size: 1rem;
  font-weight: 900;
  padding: 0.3rem;
  text-align: center;
}

.c1 {
  color: #00111f;
  padding: 0.3rem;
}

<li
  class="c0"
>
  <a
    aria-label="Next"
    class="c1"
    href="/#"
  >
    <span
      aria-hidden="true"
    >
      »
    </span>
    <span>
      Next
    </span>
  </a>
</li>
`;

exports[`PageDirection should match the snapshot when direction is = Previous 1`] = `
.c0 {
  background-color: rgba(255,255,255,1)!important;
  font-size: 1rem;
  font-weight: 900;
  padding: 0.3rem;
  text-align: center;
}

.c1 {
  color: #00111f;
  padding: 0.3rem;
}

<li
  class="c0"
>
  <a
    aria-label="Previous"
    class="c1"
    href="/#"
  >
    <span
      aria-hidden="true"
    >
      «
    </span>
    <span>
      Previous
    </span>
  </a>
</li>
`;
